crates/next-core/src/next_edge/unsupported.rs

```diff
@@ -55,20 +57,32 @@ impl ImportMappingReplacement for NextEdgeUnsupportedModuleReplacer {
     ) -> Result<Vc<ImportMapResult>> {
         let request = &*request.await?;
         if let Request::Module { module, .. } = request {
-            // packages/next/src/server/web/globals.ts augments global with
-            // `__import_unsupported` and necessary functions.
-            let code = formatdoc! {
-              r#"
-              {TURBOPACK_EXPORT_NAMESPACE}(__import_unsupported(`{module}`));
-              "#
-            };
-            let content = AssetContent::file(File::from(code).into());
-            let source = VirtualSource::new(root_path, content).to_resolved().await?;
-            return Ok(
-                ImportMapResult::Result(ResolveResult::source(ResolvedVc::upcast(source))).cell(),
-            );
-        };
-
-        Ok(ImportMapResult::NoEntry.cell())
+            // Call out to separate `unsupported_module_source` to only have a single Source cell
+            // for requests with different subpaths: `fs` and `fs/promises`.
+            let source = unsupported_module_source(root_path, module.clone())
```

Comment:
This can even futher deduplicate modules.

```suggestion
            let source = unsupported_module_source(lookup_path.root(), module.clone())
```